### PR TITLE
Added perf test and providing fix to install perf packages.

### PR DESCRIPTION
**Pull request by shriyak**
Perf test is the built in tool which executes assorted sanity tests, initially through linked routines
but also will look for a directory with more tests in the form of scripts.
Fix will install perf using linux-tools-common and linux-tools for Ubuntu distro.

Signed-off-by: Shriya <shriyak@linux.vnet.ibm.com>

### CODE REVIEW

File: perf/perftool.py [hunk 1]
```diff
@@ -39,7 +39,15 @@ def setUp(self):
 
         # Check for basic utilities
         smm = SoftwareManager()
-        for package in ['gcc', 'make', 'perf']:
+        detected_distro = distro.detect()
+        kernel_ver = platform.uname()[2]
+        if 'Ubuntu' in detected_distro.name:
+            deps = ['gcc', 'make', 'linux-tools-common',
```
Comment (PraveenPenguin) [inline perf/perftool.py]: can gcc, make package is kind pre defined in deps as it is kind of common 

ex: 
```
deps=[''gcc', 'make'']
if 'Ubuntu' in detected_distro.name:
 deps.extend(['linux-tools-common',  'linux-tools-%s' % kernel_ver ])
else:
   deps.append('perf')
```

File: perf/perftool.py [hunk 2]
```diff
@@ -51,6 +59,19 @@ def setUp(self):
...
 
     def test(self):
+        '''
+        perf test :Does sanity tests and
+        execute the tests by calling each module
+        '''
+        # Built in perf test
+        result = process.run("perf test")
+        x = result.stderr.splitlines()
+        for string in x:
```
Comment (PraveenPenguin) [inline perf/perftool.py]: can this be like 
```
    for failedString in process.run("perf test").stderr.splitlines():
           if 'FAILED' in str(failedString):
                   self.fail("Test case failed is %s", y.strip("[]")
```

just eliminate extra variables and self.log to self.fail as i think in this check we need to fail the test

Comment (PraveenPenguin) [inline perf/perftool.py]: can this be like 
```
    for failedString in process.run("perf test").stderr.splitlines():
           if 'FAILED' in str(failedString):
                   self.fail("Test case failed is %s", y.strip("[]")
```

just eliminate extra variables and self.log to self.fail as i think in this check we need to fail the test

File: perf/perftool.py [hunk 3]
```diff
@@ -39,7 +39,15 @@ def setUp(self):
...
+        kernel_ver = platform.uname()[2]
+        if 'Ubuntu' in detected_distro.name:
+            deps = ['gcc', 'make', 'linux-tools-common',
+                    'linux-tools-' + kernel_ver + '']
+        else:
+            deps = ['gcc', 'make', 'perf']
+
+        for package in deps:
             if not smm.check_installed(package) and not smm.install(package):
                 self.error('%s is needed for the test to be run' % package)
```
Comment (PraveenPenguin) [inline perf/perftool.py]: one request as you submitting changes can you please change  `self.error` to` self.skip`

File: perf/perftool.py (hunk 1 above)
Comment (shriyak) [inline perf/perftool.py]: Sure, let me make these changes.

File: perf/perftool.py (hunk 2 above)
Comment (shriyak) [inline perf/perftool.py]: Yes, let me update it

**Comment by shriyak** [comment]
Raised new PR

**Pull request closed without merging**